hugo/content/docs/learn/workflow/resolve_cross_references.md

```diff
@@ -0,0 +1,229 @@
+---
+title: "5. Resolve cross-references"
+weight: 600
+url: /docs/learn/workflow/resolve_cross_references
+---
+
+This step takes place after generating the AST. The AST definition was created and you are able to parse input files. But the AST is not complete yet. It contains _cross-references_ that are not resolved. Cross-references are used to reference other elements in your language.
+
+## Problem
+
+Let's illustrate the problem using the Hello-World example from the Yeoman generator:
+
+```text
+person John
+person Jane
+
+Hello John!
+Hello Jane!
+```
+
+The following syntax tree is generated by the Langium parser during the runtime. Mind the gaps with the question marks. These are the missing pieces you want to fill out in this step.
+
+{{<mermaid>}}
+graph TB
+  Model-->persons
+  Model-->greetings
+  
+  persons-->P1[Person]
+  P1 --> H1('person')
+  P1 --> N1[name]
+  N1 --> NL1('John')
+  
+  persons-->P2[Person]
+  P2 --> H2('person')
+  P2 --> N2[name]
+  N2 --> NL2('Jane')
+
+  greetings-->G1[Greeting]
+  G1 --> KW1('hello')
+  G1 --> PRef1[Ref]
+  PRef1 -- $refText --> RT1('John')
+  G1 --> EM1('!')
+  PRef1 --> QM1{?}
+
+  greetings-->G2[Greeting]
+  G2 --> KW2('hello')
+  G2 --> PRef2[Ref]
+  PRef2 -- $refText --> RT2('Jane')
+  G2 --> EM2('!')
+  PRef2 --> QM2{?}
+{{</mermaid>}}
+
+You normally can achieve the cross-reference resolution by implementing a so-called scope provider and a scope computation. When setup correctly given syntax tree will change to this:
+
+{{<mermaid>}}
+graph TB
+  Model-->persons
+  Model-->greetings
+  
+  persons-->P1[Person]
+  P1 --> H1('person')
+  P1 --> N1[name]
+  N1 --> NL1('John')
+  
+  persons-->P2[Person]
+  P2 --> H2('person')
+  P2 --> N2[name]
+  N2 --> NL2('Jane')
+
+  greetings-->G1[Greeting]
+  G1 --> KW1('hello')
+  G1 --> PRef1[Ref]
+  PRef1 -- $refText --> RT1('John')
+  G1 --> EM1('!')
+  PRef1 -..-> P1
+
+  greetings-->G2[Greeting]
+  G2 --> KW2('hello')
+  G2 --> PRef2[Ref]
+  PRef2 -- $refText --> RT2('Jane')
+  G2 --> EM2('!')
+  PRef2 -..-> P2
+{{</mermaid>}}
+
+## Resolution of cross-references
+
+As already hinted, you can implement a scope provider and a scope computation. Fortunately, Langium comes with default implementations for both. But eventually as your language grows, you might want to implement your own strategy because the default is not sufficient. In the follwoing sections I will sketch how to interpret their interfaces.
```

Comment:
Maybe reword this to remove the "I" from this? (Who is "I" for the documentation of an open source project?)